### ISSUE TITLE
False alarms for ClickHouse 

### ISSUE THREAD

**Issue by Hongbo-Miao** [sqlfluff/sqlfluff#4743]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

query.sql

```sql
create table cpu_metrics
(
    `name` String,
    `timestamp` datetime,
    `value` Float32
)
    engine = MergeTree
        order by (name, timestamp);

insert into cpu_metrics
select
    'cpu',
    ts + ((60 * 60) * randCanonical()) as ts,
    round(val * (0.95 + (randCanonical() / 20)), 2) as val
from (
    select
        toDateTime('2022-12-12 12:00:00') - interval k hour as ts,
        round((100 * c) / m, 2) as val
    from (
        select
            k,
            c,
            max(c) over () as m
        from (
            select
                floor(randBinomial(24, 0.5) - 12) as k,
                count(*) as c
            from numbers(1000)
            group by k
            order by k asc
        )
    )
) as a
inner join numbers(1000000) as b on 1 = 1;
```

I got this error:

```shell
sqlfluff fix --force --dialect=clickhouse query.sql

==== finding fixable violations ====
== [query.sql] FAIL
L:  13 | P:  23 | CP03 | Function names must be consistently upper case.
                       | [capitalisation.functions]
L:  14 | P:   5 | CP03 | Function names must be consistently upper case.
                       | [capitalisation.functions]
L:  14 | P:  26 | CP03 | Function names must be consistently upper case.
                       | [capitalisation.functions]
L:  16 | P:   5 | LT09 | Select targets should be on a new line unless there is
                       | only one select target.
                       | [layout.select_targets]
L:  17 | P:   9 | CP03 | Function names must be consistently upper case.
                       | [capitalisation.functions]
L:  23 | P:  13 | CP03 | Function names must be consistently upper case.
                       | [capitalisation.functions]
L:  26 | P:  17 | CP03 | Function names must be consistently upper case.
                       | [capitalisation.functions]
L:  26 | P:  23 | CP03 | Function names must be consistently upper case.
                       | [capitalisation.functions]
L:  27 | P:  17 | CP03 | Function names must be consistently upper case.
                       | [capitalisation.functions]
L:  28 | P:  18 | CP03 | Function names must be consistently upper case.
                       | [capitalisation.functions]
L:  34 | P:  12 | CP03 | Function names must be consistently upper case.
                       | [capitalisation.functions]
L:  34 | P:  32 | AL05 | Alias 'b' is never used in SELECT statement.
                       | [aliasing.unused]
  [4 templating/parsing errors found]
```

### Expected Behaviour

No errors, or auto fixed.

### Observed Behaviour

1. This is false alarm "Function names must be consistently upper case." Especially `randCanonical` cannot be uppercase, otherwise ClickHouse will throw error:
    > Error: Unknown function RANDCANONICAL. Maybe you meant: ['randCanonical','randBinomial']: While processing 'cpu', ts + ((60 * 60) * RANDCANONICAL()) AS ts, round(val * (0.95 + (randCanonical() / 20)), 2) AS val. (UNKNOWN_FUNCTION) (version 23.3.1.2823 (official build))
Error Code: 46

2. I am not sure how to fix "Select targets should be on a new line unless there is only one select target."

3. "Alias 'b' is never used in SELECT statement." is false alarm, `as b` cannot be removed. Otherwise, ClickHouse will throw error
    > Error: 500: Code: 206. DB::Exception: No alias for subquery or table function in JOIN (set joined_subquery_requires_alias=0 to disable restriction). While processing ' numbers(1000000)'. (ALIAS_REQUIRED) (version 23.3.1.2823 (official build))

### How to reproduce

Check "What Happened" section.

### Dialect

clickhouse

### Version

2.0.3

### Configuration

.sqlfluff

```
[sqlfluff]
exclude_rules = L016
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by konnectr** [comment]
Maybe excluded rule - https://docs.sqlfluff.com/en/stable/reference/rules.html#rule-CP03 ?
